realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -281,6 +283,40 @@ public void refresh() {
     }
 
     /**
+     * Blocks the current thread until transactions are committed by other threads to this Realm or
+     * {@link #stopWaitForChange()} is called from other threads.
+     *
+     * @return {@code true} if transactions are committed to this Realm instance, {@code false} if
+     * wait has been cancelled by {@link #stopWaitForChange()} called.
+     * @throws IllegalStateException if attempting to wait within a transaction.
+     */
+    public boolean waitForChange() {
+        checkIfValid();
+        if (isInTransaction()) {
+            throw new IllegalStateException("Cannot wait for changes inside of a transaction.");
+        }
+        if (handlerController != null) {
+            throw new IllegalStateException("Cannot wait for changes inside of a Looper.");
```

Comment:
"Cannot wait for changes inside a Looper Thread. Use RealmChangeListeners instead."